src/python/bot/fuzzers/syzkaller/runner.py

```diff
@@ -148,12 +167,18 @@ def _filter_log(content):
         result += strip_regex.sub('', line) + '\n'
       return result
 
+    # Download corpus ahead of fuzzing
+    download_corpus()
```

Comment:
We don't need to do this, as ClusterFuzz automatically synces corpus before fuzzing and synces it back automatically after fuzzing. All you need is to put things in 
options.corpus_dir
https://github.com/google/clusterfuzz/blob/master/src/python/bot/fuzzers/honggfuzz/engine.py#L150
and everything happens automatically.

Now you still need to put the corpus.db in the right directory and tell syzkaller to load corpus.db. Can you add code that use this corpus.db file. Also, another thing Max wanted to not sync corpus db if it is smaller than previous one, so you should save file size at start of function and then only sync back if file increased in size. we can even ignore this or put in next cl.

Reply:
Not sure i understood your comment: Is the suggestion to:
1) before fuzzing, copy corpus.db from options.corpus_dir (e.g. ~/clusterfuzz/872Y00ELP/clusterfuzz/bot/inputs/data-bundles/syzkaller) to the workdir where syzkaller will be using corpus.db i.e. ~/clusterfuzz/876X00GUE/bot/clusterfuzz/bot/inputs/fuzzer-testcases-disk/syzkaller
2) after fuzzing, copy back from workdir to options.corpus_dir?

Regarding copying using family (e.g. blueline) for device running blueline and then copying the file only if it is larger, that sounds like we will lose some corpus since even if it is smaller, it could contain different corpus. Shouldnt this be done by syzhub?

Reply:
Yes you are right.
1. before fuzzing, check if corpus.db exists in options.corpus_dir and copy to work dir.
2. after fuzzing, check if size of corpus.db in work dir is bigger than one in options.corpus_dir (if exists) and if yes, copy it to options.corpus_dir. 
rest of gcs sync back and forth is done automatically by clusterfuzz from options.corpus_dir.

Regarding your second question, what can happen is if there are two hosts running two different family of devices for syzkaller fuzzing, then their bots can have different corpus.db files in gcs. since name is the same, they will keep overriding each other's one. regarding syzhub, i dont know you are setting it per host or across multiple hosts. Also, this can be fixed later once you have more device families, for now, we can do the more straightforward cl.

Reply:
Done.